docs/src/main/sphinx/release/release-446.md

```diff
@@ -0,0 +1,67 @@
+# Release 446 (29 Apr 2024)
+
+## General
+
+* Improve performance of `INSERT` statements into partitioned tables when the
+  `retry_policy` configuration property is set to `TASK`. ({issue}`21661 `)
+* Reduce delay before stopping queries when the cluster runs out of memory. ({issue}`21719`)
+* Fix catalog names including `"` marks when creating them with `CREATE CATALOG`
+  statements. ({issue}`21399`)
+* Fix query failure when a token in a query ends with a `:`. ({issue}`21676`)
+* Fix potential query failure for queries using [SQL routines](/routines) with
+  the `LEAVE`, `ITERATE`, `REPEAT`, or `WHILE` labels. ({issue}`21682`)
+* Fix potential query failure for queries using [SQL routines](/routines) that
+  invoke other functions. ({issue}`19820`)
+* Fix potential query failure due to worker nodes running out of memory in
+  concurrent scenarios. ({issue}`21706`)
+* Fix non-propagation of OpenTelemetry tracing context when invoking a
+  connector. ({issue}`21575`)
+
+## BigQuery connector
+
+* {{breaking}} Enable `bigquery.arrow-serialization.enabled` by default.
```

Comment:
This is due to an apache arrow requirement. There is no workaround for it as far as I know. The same applies to snowflake connector which is using Arrow as well.

Reply:
The context is we want to remove another Avro serialization eventually since it's slower than Arrow and keeping both code requires maintenance costs.

Reply:
https://arrow.apache.org/docs/java/install.html#java-compatibility

Reply:
That’s a big problem. It means that those libraries are using internal JDK APIs that may disappear at any point.

Depending on such behavior may make it hard or impossible to update to new Java versions in the future. In the worst case, we may need to update to the next major Java version to fix security issues and be forced to break those connectors.

We did a lot of work when migrating to Java 11 to avoid having to use internal Java APIs to be where we are today. This is a step backwards.

Reply:
I agree but the alternative being FFM based allocator is not yet there

Reply:
https://github.com/apache/arrow/pull/38016

Reply:
I've added an FFM based AllocationManager on top of the JDK 22 PR and it works but Arrow still needs an `--add-opens` because `org.apache.arrow.memory.util.MemoryUtil` uses `Unsafe`.

Reply:
https://github.com/trinodb/trino/pull/21776

Reply:
Unsafe doesn't require add-opens. Are they using something else?

Reply:
Does Arrow use native memory? If so, how are we advising users to configure Trino when part of the host memory needs to be reserved for off-heap buffers?

Reply:
@martint yes it does and it's been problematic for some time

Reply:
Let's go ahead with the release for now and we can revisit whether we want to revert this once we have alignment in the next couple of days.

Reply:
Why do we have to give arrow native memory?  It is impossible for it to use heap memory?  Is this code pulling in native libraries also?

Reply:
Actually, we fork arrow to use heap memory.  I have no idea why they think they need native memory

Reply:
@dain native - no. ArrowBuf allocates and uses off heap memory. The idea is to allow to pass vectors to other languages but that's not an actual case in these connectors